### ISSUE TITLE
Method covering a forwarded method does not generate an offense

### ISSUE THREAD

**Issue by toddsiegel** [rubocop-hq/rubocop#3119]
I have the following code that does not generate a lint error, but should IMO.

If you agree, I am happy to work on this with a little guidance on whether this should be its own cop or thrown in with `Lint/DuplicateMethods`.

``` ruby
require 'forwardable'

class Inner
  def foo
    puts 'from inner'
  end
end

class Outer
  extend Forwardable

  def_delegators :@inner, :foo

  def initialize(inner)
    @inner = inner
  end

  def foo
    'from outer'
  end
end
```
## Expected behavior

Since it's basically like defining `#foo` twice in `Outer`, which would be a `Lint/DuplicateMethods` error, I think this should be an error too.
## Actual behavior

Does not generate an error.
## Steps to reproduce the problem

Run the above code through RuboCop, e.g. `rubocop dupmethod.rb`
## RuboCop version

```
$ rubocop -V
0.40.0 (using Parser 2.3.1.0, running on ruby 2.2.3 x86_64-darwin14)
```
